src/main/java/org/apache/sysml/hops/codegen/opt/PlanSelectionFuseCostBasedV2.java

```diff
@@ -215,20 +215,23 @@ private void selectPlans(CPlanMemoTable memo, PlanPartition part)
 	 * @return optimal assignment of materialization points
 	 */
 	private static boolean[] enumPlans(CPlanMemoTable memo, PlanPartition part, StaticCosts costs, 
-		ReachabilityGraph rgraph, InterestingPoint[] matPoints, int off, double bestC) 
+		ReachabilityGraph rgraph, InterestingPoint[] matPoints, int off, double bestC)
 	{
 		//scan linearized search space, w/ skips for branch and bound pruning
 		//and structural pruning (where we solve conditionally independent problems)
 		//bestC is monotonically non-increasing and serves as the upper bound
-		long len = (long)Math.pow(2, matPoints.length-off);
+		long len = 1L << matPoints.length-off;
```

Comment:
I wrote a micro-benchmark and measured that using a left shift is over 7x faster than using `Math.pow`. I personally find it just as readable.

```java
int limit = 50_000_000;
@Test
public void testPower() {
	Random r = new Random(7);
	long t = System.currentTimeMillis();
	for (int i = 0; i < limit; i++) {
		int p = r.nextInt(63);
		long l = (long)Math.pow(2,p);
	}
	long t1 = System.currentTimeMillis();
	System.out.println((t1-t)/1000.0); // 3.758
}
@Test
public void testShift() {
	Random r = new Random(7);
	long t = System.currentTimeMillis();
	for (int i = 0; i < limit; i++) {
		int p = r.nextInt(63);
		long l = 1L << p;
	}
	long t1 = System.currentTimeMillis();
	System.out.println((t1-t)/1000.0); // 0.523
}
```